x-pack/plugins/security/public/account_management/user_profile/user_profile.tsx

```diff
@@ -639,12 +739,59 @@ export function useUserProfileForm({ user, data }: UserProfileProps) {
         return;
       }
 
+      let isRefreshRequired = false;
+      if (initialValues.data?.userSettings.darkMode !== values.data?.userSettings.darkMode) {
+        isRefreshRequired = true;
+      }
```

Comment:
There's a slight issue with this logic in that we will end up requiring a page refresh when e.g. changing from "default" to "light" when the default is "light". 

Given that it's probably not trivial to read out what the default actually is it's not the end of the world. 

Otherwise the flow feels very slick and seamless 👌

Reply:
That is a great point; I could try to smooth this out maybe in a future issue (this change could probably aide in a 'System/OS Theme' option)